x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportStopRollupAction.java

```diff
@@ -52,12 +57,52 @@ protected void taskOperation(StopRollupJobAction.Request request,
                                  ActionListener<StopRollupJobAction.Response> listener) {
         if (jobTask.getConfig().getId().equals(request.getId())) {
             jobTask.stop(listener);
+            waitForStopped(request, jobTask, listener);
```

Comment:
We shouldn't block the current thread. I think we can use the `generic` thread pool here in order to make sure that we don't block on important threads (network threads). Something like:
````
 threadPool.generic().execute(new AbstractRunnable() {
                   @Override
                    protected void doRun() throws Exception {
                          if (waitForStopped(request, jobTask)) {
                            listener.onResponse(new Stop...);
                          } else {
                            listener.onFailure(new Stop...);
                         }
                    }

                    @Override
                    public void onFailure(Exception e) {
                        listener.onFailure(e);
                    }

Reply:
Should we just have the task run on the Generic thread from the start?  E.g. in the ctor just specify `generic` instead of `same`?

Reply:
I prefer if we explicitly fork to another thread with a comment explaining why we do that.  But yes I think it would work if we use `generic` directly. No strong opinion here either, just a preference ;)

Reply:
++